iceoryx-sys/src/subscriber.rs

```diff
@@ -244,29 +244,55 @@ impl Subscriber {
         }
     }
 
-    pub fn get_chunk<T>(&self) -> Option<*const T> {
+    pub fn try_take<T>(&self) -> Option<RawSample<T>> {
+        unsafe { self.try_get_chunk().map(|payload| payload.cast::<T>()) }
+    }
+
+    pub fn try_take_slice<T>(&self) -> Option<RawSample<[T]>> {
         unsafe {
-            let payload = cpp!([self as "SubscriberPortUser*"] -> *const std::ffi::c_void as "const void*" {
-                auto getChunkResult = self->tryGetChunk();
+            let payload = self.try_get_chunk()?;
 
-                if (getChunkResult.has_error()) {
-                    return nullptr;
-                }
+            let chunk_header = payload.chunk_header();
+            let payload_size = chunk_header.get_user_payload_size();
```

Comment:
could you `usize` as `payload_size` type right here with `as usize` afterwards. It would make the if statement below easier to read when there is no cast in between the conditions. The same goes for `payload_alignment`.